mesheryctl/internal/cli/root/models/import.go

```diff
@@ -0,0 +1,266 @@
+		if err != nil {
+			return err
+		}
+		ctx, err := mctlCfg.GetCurrentContext()
+		if err != nil {
+			return system.ErrGetCurrentContext(err)
+		}
+		err = ctx.ValidateVersion()
+		if err != nil {
+			return err
+		}
+		return nil
+	},
+	Args: func(cmd *cobra.Command, args []string) error {
+		const errMsg = "Usage: mesheryctl exp model import <URI/path to Model>\nRun 'mesheryctl exp model import --help' to see detailed help message"
+		if len(args) == 0 {
+			if err := cmd.Usage(); err != nil {
+				return err
+			}
+			return utils.ErrInvalidArgument(errors.New(errMsg))
+		}
+		return nil
+	},
+
+	RunE: func(cmd *cobra.Command, args []string) error {
+		if validURL := govalidator.IsURL(args[0]); !validURL {
+			return importFromLocalFile(args[0])
+		} else if isGithubURL := strings.Contains(args[0], "github.com"); isGithubURL {
+			return importFromGithubURL(args[0])
+		}
+		return nil
+	},
+}
+
+func importFromLocalFile(filePath string) error {
+	if _, err := os.Stat(filePath); os.IsNotExist(err) {
+		return utils.ErrFileRead(errors.Errorf("File path does not exist. Path provided: %s", filePath))
+	}
+
+	if strings.HasSuffix(filePath, ".zip") {
+		return processZip(filePath)
+	}
+
+	if strings.HasSuffix(filePath, ".tar.gz") || strings.HasSuffix(filePath, ".tgz") {
+		return processTarGz(filePath)
+	}
+
+	dirPath := filePath
+	// fmt.Println("Directory path:", dirPath)
+	return processDirectory(dirPath)
+}
+
+func processZip(zipFilePath string) error {
+	homeDir, _ := os.UserHomeDir()
+	tmpDir := homeDir + "/.meshery/temp"
+	if err := os.MkdirAll(tmpDir, 0755); err != nil {
+		return system.ErrCreateDir(err, tmpDir)
+	}
+	defer os.RemoveAll(tmpDir)
+
+	if err := extract.ExtractZip(tmpDir, zipFilePath); err != nil {
+		return utils.ErrExtractFile(err, "zip")
+	}
+
+	return processDirectory(tmpDir)
+}
+
+func processTarGz(tarGzFilePath string) error {
+	homeDir, _ := os.UserHomeDir()
+	tmpDir := homeDir + "/.meshery/temp"
+	if err := os.MkdirAll(tmpDir, 0755); err != nil {
+		return system.ErrCreateDir(err, tmpDir)
+	}
+	defer os.RemoveAll(tmpDir)
+
+	if err := extract.ExtractTarGz(tmpDir, tarGzFilePath); err != nil {
+		return utils.ErrExtractFile(err, "tar.gz")
+	}
+
+	return processDirectory(tmpDir)
+}
+
+func processDirectory(dirPath string) error {
+	entries, err := os.ReadDir(dirPath)
+	if err != nil {
+		return utils.ErrDirRead(err, dirPath)
+	}
+	imported := true 
+	for _, entry := range entries {
+		if entry.IsDir() {
+			subdirPath := filepath.Join(dirPath, entry.Name())
+			err := processDirectory(subdirPath)
+			if err != nil {
+				continue
+			}
+		} else {
+			fileContent, err := os.ReadFile(filepath.Join(dirPath, entry.Name()))
+			if err != nil {
+				continue
+			}
+			if err := registerComponent(entry.Name(), fileContent); err != nil {
+				// utils.Log.Info("Error registering component: ", entry.Name(), err)
+				imported = false
+				break
+			}
+		}
+	}
+	if imported {
+		utils.Log.Info("Model imported successfully")
```

Comment:
Ensure this message only occurs once. Don't say "successfully". Verify the count of components, and relationships. 

Return the model name / id / command to view the model; to list the components/relationships